src/Contrib/StackSdks/Masa.Contrib.StackSdks.Tsc.Elasticsearch/TraceService.cs

```diff
@@ -27,8 +27,10 @@ public Task<PaginatedListBase<TraceResponseDto>> ListAsync(BaseRequestDto query)
         return _client.SearchTraceAsync(query);
     }
 
-    public void GetAll(BaseRequestDto query, Action<IEnumerable<TraceResponseDto>> resultAction)
+    public async Task<PaginatedListBase<TraceResponseDto>> ScrollAsync(BaseRequestDto query)
     {
-        _client.ScrollQuery((ElasticsearchScrollRequestDto)query, resultAction);
+        if (query is not ElasticsearchScrollRequestDto)
+            throw new UserFriendlyException("parameter: query must is type: ElasticsearchScrollRequestDto");
+        return await _client.SearchTraceAsync(query);
```

Comment:
It can be considered that await is not applicable and returns directly to Task<PaginatedListBase<TraceResponseDto>>